__tests__/api.test.js

```diff
@@ -0,0 +1,39 @@
+const db = require('../db/connection');
+const data = require('../db/data/test-data');
+const seed = require('../db/seeds/seed');
+const request = require("supertest");
+const app = require('../db/app');
+
+beforeEach(() => seed(data));
+afterAll(() => db.end());
+
+describe('GET /api/topics', () => {
+    test('Responds with status code 200 and with response message of all topics', () => {
+    return request(app)
+        .get('/api/topics')
+        .expect(200)
+        .then(({ body }) => {
+            expect(body).toHaveProperty('topics');
+            });
+    });  
+    test("Data is in the correct format", () => {
+        return request(app)
+        .get('/api/topics')
+        .then(({ body }) => {
+        const topics = body.topics;
+        expect(Array.isArray(topics)).toBe(true);
+        topics.forEach((topic) => {
+            expect(typeof topic.slug).toBe('string');
+            expect(typeof topic.description).toBe('string');            
+        });
+        });
```

Comment:
Nice looping through with forEach, make sure you also check for the length of the array here. You can also use the following syntax to check key value pairs - 'expect(topic).toHaveProperty('slug', expect.any(String))'

Reply:
checked for the length of the array in the tests